deploying-monitoring-redis-oke.md

```diff
@@ -0,0 +1,257 @@
+
+### Mass Insert Data into Redis
+Let’s now do a mass insertion of data into Redis. I found this neat gem to load a csv file into redis.
+
+Given a csv file of the following format:
+
+```
+id, first name, age, gender, nickname, salary
+1, John Smith, 40, Male, John, 10000
+2, Marco Polo, 43, Male, Marco, 10000
+….
+1999999, Tom Cruse, 50, Male, Tom, 10001
+```
+
+The following command can be used to import that csv file into Redis:
+
+```console
+awk -F, 'NR > 1{ print "SET", "\"employee_"$1"\"", "\""$0"\"" }' file.csv | redis-cli --pipe
+```
+
+First, we have to generate the dataset. We will be using the mimesis package:
+
+```console
+pip install mimesis
+```
+
+And we will adapt the schema a little bit so we can make use of whatever mimesis provides to create a csv file using Python:
+
+<!--EDIT: This appears to be mangled python, please validate from source-->
+```console
+import csvfrom mimesis import Personfrom mimesis.enums import Genderen = Person('en')with open('file.csv',mode='w') as csv_file:field_names = ['id', 'full name', 'age', 'gender', 'username', 'weight']writer = csv.DictWriter(csv_file, fieldnames=field_names)writer.writeheader()for n in range(100000):writer.writerow({'id': str(n), 'first name': en.full_name(), 'age': str(en.age()), 'gender': en.gender(), 'username':en.username(), 'weight':str(en.weight())})
+```
+
+Run the Python script to generate the data:
+
+```console
+python names.py
+```
+
+This will create a file.csv in the current directory. You can [configure a PersistentVolume](https://kubernetes.io/docs/tasks/configure-pod-container/configure-persistent-volume-storage/) to store and load the data but for the purpose of this exercise, we'll do a quick hack by installing redis in the bastion:
+
+```console
+sudo yum install redis -y
+```
+
+This will allow us to use the redis-cli from the bastion where we have generated/uploaded the file.csv.
+
+On the bastion, get a list of Redis pods:
+
+```console
+kubectl -n redis get pods
+NAME                             READY   STATUS    RESTARTS   AGE                                                                                
+redis-master-0                   1/1     Running   0          156m                                                                               
+redis-metrics-794db76ff7-xmd2q   1/1     Running   0          156m                                                                               
+redis-slave-7fd8b55f7-25w8d      1/1     Running   1          156m                                                                               
+redis-slave-7fd8b55f7-hvhmc      1/1     Running   1          156m                                                                               
+redis-slave-7fd8b55f7-mjq8q      1/1     Running   1          156m
+```
+
+Afterwards, use port-forward to so you can access it using the redis-cli:
+
+<!--EDIT is there actually a command `k`?-->
```

Comment:
There might be a command `k`, but I'm assuming there's something missing. Please validate.

Reply:
Nah, this one is actually exactly faithful to the code block in the article, formatting and all. `k` isn't an option/flag as far as I can tell.